restcomm/restcomm.telephony/src/main/java/org/restcomm/connect/telephony/CallManager.java

```diff
@@ -2272,7 +2207,8 @@ public void response(final Object message) throws IOException {
 
         // if this response is coming from a client that is in a p2p session with another registered client
         // we will just proxy the response
-        if (B2BUAHelper.isB2BUASession(response)) {
+        SipSession linkedB2BUASession = B2BUAHelper.getLinkedSession(response);
+        if (linkedB2BUASession!=null) {
```

Comment:
i take this is actually the only change to fix the issue

Reply:
`B2BUAHelper.isB2BUASession(response)` basically checks for the existence of a session in the message. 
Since the block previously did not need to access any session data, it did not retrieve the `SipSession`. Now we do need to one access session attribute `EXTENSION_HEADERS` so we do need the local `SipSession`